File: backend/app/api/routes/activities.py
from datetime import datetime

from fastapi import APIRouter, Depends

from app.api.deps import get_current_user_id
from app.schemas.trips import Trip

router = APIRouter(prefix="/activities")


@router.get("/dining/events", response_model=list[Trip])
def get_dining_events(_user_id: str = Depends(get_current_user_id)) -> list[Trip]:
    return []


@router.get("/movies/events", response_model=list[Trip])
def get_movies_events(_user_id: str = Depends(get_current_user_id)) -> list[Trip]:
    return []


@router.get("/play/events", response_model=list[Trip])
def get_play_events(_user_id: str = Depends(get_current_user_id)) -> list[Trip]:
    return []

File: lib/api/activities.ts
import { apiRequest } from './client';
import { Trip } from './types';

export function getDiningEvents(_userId: string) {
    return apiRequest<Trip[]>('/activities/dining/events', { method: 'GET' });
}

export function getMovieEvents(_userId: string) {
    return apiRequest<Trip[]>('/activities/movies/events', { method: 'GET' });
}

export function getPlayEvents(_userId: string) {
    return apiRequest<Trip[]>('/activities/play/events', { method: 'GET' });
}
